src/pathfinder/yapf/yapf_rail.cpp

```diff
@@ -446,6 +446,11 @@ class CYapfFollowRailT : public CYapfReserveTrack<Types>
 
 				this->FindSafePositionOnNode(pPrev);
 			}
+
+			/* There is no next track if the origin node is also a destination node.
+			 * Note that we still consider the path to be found, this is on purpose. */
```

Comment:
This sentence is hard to decipher, and I am not sure what you are trying to say. The return indicates something invalid, and the comment says "the path to be found".

Mind rewording it a bit to make more clear what you are trying to say? :D

Reply:
See my updated commit, I hope this clears things up a bit :)